### PR TITLE
Refactor of mixinAttach generator

### PR DESCRIPTION
**Pull request by jzaffiro**
Updates the generatorFactory in mixinAttach to perform a chunk of ops, attach, then complete the rest of the ops generated. The new format will always generate an attach op, instead of using a probability to determine whether or not to generate an attach.  

AB#6447

### CODE REVIEW

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 1]
```diff
@@ -326,6 +328,7 @@ export interface DDSFuzzSuiteOptions {
 	detachedStartOptions: {
 		attachProbability: number;
```
Comment (anthony-murphy) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:329]: attachProbability should be removed

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 2]
```diff
@@ -465,6 +468,7 @@ export const defaultDDSFuzzSuiteOptions: DDSFuzzSuiteOptions = {
 	detachedStartOptions: {
 		attachProbability: 0.2,
 		enabled: true,
+		numOpsBeforeAttach: 0,
```
Comment (anthony-murphy) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:471]: i think the default should be more than 0, as with the previous mechanism, attachProbability, we would do some

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 3]
```diff
@@ -324,8 +326,8 @@ export interface DDSFuzzSuiteOptions {
 	 * an attach op.
 	 */
 	detachedStartOptions: {
-		attachProbability: number;
 		enabled: boolean;
+		numOpsBeforeAttach: number;
```
Comment (Abe27342) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:330]: I think the scenarios that `enabled: false` powered before can be replaced with `numOpsBeforeAttach: 0`. Might be a nice simplification of the API.

File: packages/dds/task-manager/src/test/taskManager.fuzz.spec.ts [hunk 4]
```diff
@@ -275,8 +275,7 @@ describe("TaskManager fuzz testing with rebasing", () => {
 		saveFailures: { directory: path.join(__dirname, "../../src/test/results") },
 		// AB#5341: enabling 'start from detached' within the fuzz harness demonstrates eventual consistency failures.
 		detachedStartOptions: {
-			enabled: false,
-			attachProbability: 0.2,
+			numOpsBeforeAttach: 0,
```
Comment (anthony-murphy) [inline packages/dds/task-manager/src/test/taskManager.fuzz.spec.ts:278]:
```suggestion
```

File: packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts [hunk 5]
```diff
@@ -525,19 +521,17 @@ describe("DDS Fuzz Harness", () => {
 					["A", "B", "C"],
 				);
 				assert.equal(finalState.summarizerClient.channel.id, "summarizer");
-				assert.deepEqual(generatedOperations[0], { type: "attach" });
+				assert.deepEqual(generatedOperations[0], { type: "noop" });
```
Comment (anthony-murphy) [inline packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts:524]:
```suggestion
				assert.deepEqual(generatedOperations[5], { type: "attach" });
```

Comment (anthony-murphy) [reply packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts:524]: this should verify an attach off is generated when it is expected to

**Review by anthony-murphy — APPROVED** [review]

**Comment by anthony-murphy** [comment]
i took a look at your test failures here, looks like they are for 0x54e. there are currently a bunch of existing disables for that, so you'll probably need to reset those, as the overall patterns of the will change going from attachProbability to a fixed number.

**Pull request merged**